llvm/lib/Analysis/InlineCost.cpp

```diff
@@ -2260,6 +2271,48 @@ bool CallAnalyzer::simplifyCallSite(Function *F, CallBase &Call) {
   return false;
 }
 
+bool CallAnalyzer::isLoweredToCall(Function *F, CallBase &Call) {
+  // Calls to memcpy with a known constant size 1/2/4/8 should not incur a call
+  // penalty, as the calls will be folded away by InstCombine. This is only
+  // really relevant on platforms whose headers redirect memcpy to __memcpy_chk
+  // (e.g. Mac), as other platforms use memcpy intrinsics, which are already
+  // exempt from the call penalty.
+  if (GetTLI) {
+    auto TLI = GetTLI(*F);
+    LibFunc LF;
+    if (TLI.getLibFunc(*F, LF) && TLI.has(LF)) {
```

Comment:
Perhaps something like below, which would need to duplicate the fallback

```suggestion
TargetLibraryInfo *TLI = GetTLI ? &GetTLI(*F) : nullptr;
LibFunc LF;
if  (!TLI || !TLI->getLibFunc(*F, LF) || !TLI->has(LF) )
    return TI.isLoweredToCall(F);
```

Reply:
I can't say I love the duplication, but OK, done.